src/main/java/edu/harvard/iq/dataverse/api/AbstractApiBean.java

```diff
@@ -390,31 +392,48 @@ protected DataverseLinkingDataverse findDataverseLinkingDataverseOrDie(String da
     }
 
     protected Dataset findDatasetOrDie(String id) throws WrappedResponse {
+        return findDatasetOrDie(id, false);
+    }
+
+    protected Dataset findDatasetOrDie(String id, boolean deep) throws WrappedResponse {
+        Long datasetId;
         Dataset dataset;
         if (id.equals(PERSISTENT_ID_KEY)) {
             String persistentId = getRequestParameter(PERSISTENT_ID_KEY.substring(1));
             if (persistentId == null) {
                 throw new WrappedResponse(
                         badRequest(BundleUtil.getStringFromBundle("find.dataset.error.dataset_id_is_null", Collections.singletonList(PERSISTENT_ID_KEY.substring(1)))));
             }
-            dataset = datasetSvc.findByGlobalId(persistentId);
-            if (dataset == null) {
-                throw new WrappedResponse(notFound(BundleUtil.getStringFromBundle("find.dataset.error.dataset.not.found.persistentId", Collections.singletonList(persistentId))));
+            GlobalId globalId;
+            try {
+                globalId = PidUtil.parseAsGlobalID(persistentId);
+            } catch (IllegalArgumentException e) {
+                throw new WrappedResponse(
+                    badRequest(BundleUtil.getStringFromBundle("find.dataset.error.dataset.not.found.bad.id", Collections.singletonList(persistentId))));
+            }
+            datasetId = dvObjSvc.findIdByGlobalId(globalId, DvObject.DType.Dataset);
```

Comment:
Skipping the call to datasetSvc.findByGlobalId(String) changes the functionality - that call does a lookup both by the PID and the alternatePID. I think we still want that.

Reply:
I added the alt global id lookup.